handler/incoming_instance_handler.go

```diff
@@ -263,6 +273,9 @@ func (hdlr *InstanceEventHandler) setOrderAndNodeIDs(ctx context.Context, instan
 		updates = append(updates, u)
 	}
 
+	// Send a patch to dataset api with all the updates in a single call
+	// so that the mongodb lock will be acquired only once per batch.
+	// The reason is that releasing a lock has been observed in 'develop' environment to take over 40ms.
```

Comment:
Please can you change this line to:
// The reason is that releasing a lock has been observed in 'develop' environment to take about 40 or more milliseconds.